src/listener.js

```diff
@@ -0,0 +1,15 @@
+export const createDeferredListener = () => {
```

Comment:
[No blocker, suggestion] Do you think it might be worth adding some unit tests to the createDeferredListener?

Reply:
Good idea, added a test